src/qtcore/qml/QObject.js

```diff
@@ -26,5 +29,7 @@ function QObject(parent) {
 
         if (this.$parent && this.$parent.$tidyupList)
             this.$parent.$tidyupList.splice(this.$parent.$tidyupList.indexOf(this), 1);
+
+        this.parent = undefined; // must do this. => 1) parent will be notified and erase object from it's children. 2) DOM node will be removed.
```

Comment:
@pavelvasev Does that mean that it would be unneeded there after this is merged?